### ISSUE TITLE
Something went wrong, exit now..

### ISSUE THREAD

**Issue by workingmanrob** [suuhm/raspi_mpeg_license_patch.sh#5]
```
     ____________________________________________________ 
    |><><><><><><><><><><><><><><><><><><><><><><><><><><|
    |  raspi_mpeg_license_patch v0.4b - (c)2022 suuhm    |
    |____________________________________________________|
                                                          

[*] First check mpeg2/vc activated?
MPG2=disabled
WVC1=disabled

[+] Getting Hexstring... May take some time...

[!] Not patched (0x) continue...

[*] set up patch now () ...
bash: line 172: [: -eq: unary operator expected
bash: line 183: [: -eq: unary operator expected
[!] Something went wrong, exit now..
```
Another new string maybe? I'm not sure how I would find that..

This is on a pi0 but I had the same error on a pi2 as well.

**Comment by workingmanrob** [comment]
Looking at the two line numbers mentioned it must be the HS_MODE check that is failing.

if [ $HS_MODE -eq 3 ]
elif [ $HS_MODE -eq 4 ]

Added a little echo check (echo "HS_MODE is $HS_MODE") just before those lines and got:

[*] set up patch now () ...
HS_MODE is 

HTH!

**Comment by suuhm** [comment]
Hi, 
first: which command did you run? and which OS did you use for running my script?

Can you please offer the output from:

```
xxd $START_ELF | grep -Ei "47 *E9 *3(3|4) *36 *32 *48 *(3C|1D) *(18|1F)"
```

Replace here $START_ELF with your start.elf

**Comment by workingmanrob** [comment]
I wrote a big long reply that for some reason is not here.

Short version - bootloader has been updated.  String is not found on any of my updated systems.  I have OSMC and raspbian but OSMC worked until the update from this month.  I was working on raspbian when it threw the error.

`wget https://raw.githubusercontent.com/suuhm/raspi_mpeg_license_patch.sh/main/raspi_mpeg_license_patch.sh -qO- | sudo bash -s -- --patch-now --os=raspbian`
